src/Simplify.cpp

```diff
@@ -5828,6 +5829,13 @@ void simplify_test() {
         check(e, e);
     }
 
+    // This expression is used to cause infinite recursion.
+    {
+        Expr e = Broadcast::make(-16, 2) < (ramp(Cast::make(UInt(16), 7), Cast::make(UInt(16), 11), 2) - Broadcast::make(1, 2));
```

Comment:
Can we make a test for this issue that isn't a vector expression? It would be a lot simpler and get the point across much more easily. If this problem doesn't occur for non-vector expressions, that would be interesting as well.